http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ChannelPipelineInitializer.java

```diff
@@ -64,4 +68,14 @@ public void channelCreated(Channel ch) throws Exception {
         // Disabling auto-read is needed for backpressure to work
         ch.config().setOption(ChannelOption.AUTO_READ, false);
     }
+
+    @Override
+    public void channelReleased(Channel ch) throws Exception {
```

Comment:
What's the cost to re-upping these every time?

Reply:
This should happen in the existing case, this isn't new logic - it's just that we might be missing them if we get an exception. They are non "sharable" handlers, and so should only live for a the life of a single request.